### PR TITLE
Bugfix/keybind overlap

### LINKED ISSUE #13994: [Bug]: Keybindings with CTRL,ALT combination not working properly
### Expected behavior

The combination of modifiers (<kbd>ALT,CTRL</kbd>) in keybindings would work without overlapping

As default, for example:
<kbd>ALT + O</kbd> to open `Search Object`
<kbd>CTRL + ALT + O</kbd> to open `HTTP Errors`

### Actual behavior

<kbd>CTRL + ALT + O</kbd> opens both `HTTP Errors` and `Search Object` at the same time

### Steps to reproduce

After login, press this hotkey combination <kbd>CTRL + ALT + O</kbd> or any combination that exists with <kbd>ALT + something</kbd> and <kbd>CTRL + ALT + something</kbd>

### PR DESCRIPTION
**Pull request by torqdev**
## Changes in this pull request  
Resolves #13994

### CODE REVIEW

File: models/User.php [hunk 1]
```diff
@@ -990,6 +990,28 @@ public function getKeyBindings()
         return $this->keyBindings ? $this->keyBindings : self::getDefaultKeyBindings();
     }
 
+    /**
+     * @param array|null $bindings
+     * @return array|null
+     */
+    public static function strictKeybinds($bindings)
```
Comment (jdreesen) [inline models/User.php]:
```suggestion
    public static function strictKeybinds(?array $bindings): ?array
```

File: models/User.php [hunk 2]
```diff
@@ -990,6 +990,28 @@ public function getKeyBindings()
...
+    {
+        if($bindings == null){
```
Comment (jdreesen) [inline models/User.php]:
```suggestion
        if($bindings === null){
```

File: models/User.php [hunk 3]
```diff
@@ -990,6 +990,28 @@ public function getKeyBindings()
...
+            if(!array_key_exists('ctrl', $binding)){
+                $bindings[$ind]['ctrl'] = false;
+            }
+            if(!array_key_exists('alt', $binding)){
+                $bindings[$ind]['alt'] = false;
+            }
+            if(!array_key_exists('shift', $binding)){
+                $bindings[$ind]['shift'] = false;
+            }
+        }
```
Comment (jdreesen) [inline models/User.php]: Fancy modern code:

```suggestion
        foreach (array_keys($bindings) as $idx) {
            $bindings[$idx]['ctrl'] ??= false;
            $bindings[$idx]['alt'] ??= false;
            $bindings[$idx]['shift'] ??= false;
        }

```

The difference is, that `null` will be turned to `false` while it stays `null` in your solution. Does that matter?

File: models/User.php [hunk 4]
```diff
@@ -990,6 +990,28 @@ public function getKeyBindings()
...
 
+    /**
+     * @param array|null $bindings
+     * @return array|null
+     */
+    public static function strictKeybinds($bindings)
+    {
+        if($bindings == null){
+            return null;
+        }
```
Comment (jdreesen) [inline models/User.php]: I don't think we need to support the `null` case here.

```suggestion
    public static function strictKeybinds(array $bindings): array
    {
```

File: models/User.php (hunk 3 above)
Comment (cameronfromtorq) [reply models/User.php]: just checked it out and this will still work great, thanks!

File: models/User.php [hunk 5]
```diff
@@ -990,6 +990,15 @@ public function getKeyBindings()
         return $this->keyBindings ? $this->keyBindings : self::getDefaultKeyBindings();
     }
 
+    public static function strictKeybinds(array $bindings): array
```
Comment (jdreesen) [inline models/User.php:998]: Improved types (for @blankse - see #13996):

```suggestion
    /**
     * @param list<array{action: string, key: int, alt?: bool, ctrl?: bool, shift?: bool}> $bindings
     *
     * @return list<array{action: string, key: int, alt: bool, ctrl: bool, shift: bool}>
     */
    public static function strictKeybinds(array $bindings): array
```

**Comment by mcop1** [comment]
Thank you!

**Pull request merged**